engine/lib/phx/src/physics/physics.rs

```diff
@@ -1,175 +1,525 @@
+            }
+        }
+    }
+
+    pub fn sphere_overlap(&self, sphere: &Sphere) -> bool {
+        self.shape_overlap(
+            &rp::Ball {
+                radius: sphere.r as rp::Real,
+            },
+            sphere.p,
+            Quat::IDENTITY,
+        )
+    }
+
+    pub fn box_overlap(&self, pos: &Vec3, rot: &Quat, half_extents: &Vec3) -> bool {
+        self.shape_overlap(
+            &rp::Cuboid {
+                half_extents: half_extents.to_na(),
+            },
+            *pos,
+            *rot,
+        )
+    }
+
+    pub fn draw_bounding_boxes_local(&self) {}
+
+    pub fn draw_bounding_boxes_world(&self) {}
+
+    pub fn draw_triggers(&self) {}
+
+    pub fn draw_wireframes(&mut self) {
+        let world = self.world.as_ref();
+        self.debug_renderer.render(
+            &mut RapierDebugRenderer,
+            &world.rigid_bodies,
+            &world.colliders,
+            &self.impulse_joints,
+            &self.multibody_joints,
+            &world.narrow_phase,
+        )
+    }
 }
 
-#[no_mangle]
-pub unsafe extern "C" fn Physics_DrawBoundingBoxesLocal(this: &mut Physics) {
-    _cppPhysics_DrawBoundingBoxesLocal(this)
+impl Physics {
+    /// Returns a list of all rigid bodies that are contained within the shape
+    /// at the given position and rotation.
+    fn shape_cast(&self, shape: &dyn rp::Shape, pos: Vec3, rot: Quat) -> Vec<*mut RigidBody> {
+        let rp_transform =
+            rp::Isometry::from_parts(rp::Translation::from(pos.to_na()), rot.to_na());
+        let world = self.world.as_ref();
+
+        // Trigger scene query and populate results.
+        let mut result: Vec<*mut RigidBody> = vec![];
+        self.query_pipeline.intersections_with_shape(
+            &world.rigid_bodies,
+            &world.colliders,
+            &rp_transform,
+            shape,
+            rp::QueryFilter::default(),
+            |handle| {
+                if let Some(rigid_body) = RigidBody::linked_with_collider_mut(world.get(handle)) {
+                    result.push(rigid_body);
+                }
+                true
+            },
+        );
+
+        result
+    }
+
+    /// Returns true if any rigid bodies are contained within the shape at the
+    /// given position and rotation.
+    fn shape_overlap(&self, shape: &dyn rp::Shape, pos: Vec3, rot: Quat) -> bool {
+        let rp_transform =
+            rp::Isometry::from_parts(rp::Translation::from(pos.to_na()), rot.to_na());
+        let world = self.world.as_ref();
+        self.query_pipeline
+            .intersection_with_shape(
+                &world.rigid_bodies,
+                &world.colliders,
+                &rp_transform,
+                shape,
+                rp::QueryFilter::default(),
+            )
+            .is_some()
+    }
 }
 
-#[no_mangle]
-pub unsafe extern "C" fn Physics_DrawBoundingBoxesWorld(this: &mut Physics) {
-    _cppPhysics_DrawBoundingBoxesWorld(this)
+/// Convert HSL color to RGB
```

Comment:
Maybe it's better to move this to Color struct as you have suggested before?

Reply:
I'd rather leave this for https://github.com/Limit-Theory-Redux/ltheory/pull/212 - as that's the PR that introduces the colour class.

Reply:
Ah, yeah. I forgot color is not merged yet.